whois-api/src/main/java/net/ripe/db/whois/api/rdap/RedactionObjectMapper.java

```diff
@@ -1,29 +1,36 @@
 package net.ripe.db.whois.api.rdap;
 
-import com.google.common.collect.Lists;
 import com.google.common.collect.Sets;
 import net.ripe.db.whois.api.rdap.domain.Entity;
 import net.ripe.db.whois.api.rdap.domain.RdapObject;
 import net.ripe.db.whois.api.rdap.domain.Redaction;
+import net.ripe.db.whois.common.domain.CIString;
 import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class RedactionObjectMapper {
 
-    public static final List<AttributeType> REDACTED_PERSONAL_ATTR = Lists.newArrayList(AttributeType.NOTIFY,
-            AttributeType.E_MAIL);
     public static void mapRedactions(final RdapObject rdapObject) {
 
-        addEntityRedaction(rdapObject);
+        rdapObject.getRedacted().addAll(getRedactions(rdapObject.getRedactedRpslAttrs(), "$"));
 
         if(rdapObject.getNetwork() != null) {
-            addEntitiesRedaction(rdapObject, rdapObject.getNetwork().getEntitySearchResults(), "$.network");
+            final String prefix = "$.network";
```

Comment:
could we create again the method addEntityRedaction which will call addEntitiesRedaction
and move
rdapObject.getRedacted().addAll(getRedactions(rdapObject.getNetwork().getRedactedRpslAttrs(), prefix));
inside that method?

Then we could just put addEntityRedaction(rdapObject, "$.network");
Then we could just put addEntityRedaction(rdapObject, "$.networks[?(@.handle=='%s')]");

I had to debug to see what line 25 is doing, basically is create the redaction of the main entity.

In this case it is clear where we create a redaction for whole entity and when we don't. for me the code is much easier on that way

Reply:
Not really, Its a single line so I do not  see a need for putting in a method also not every entity calls entities.

Reply:
I think that it is much readable to be honest. But up to you